pxemgr/pxemanager_test.go

```diff
@@ -0,0 +1,156 @@
+package pxemgr
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/giantswarm/mayu-infopusher/machinedata"
+	"github.com/giantswarm/mayu/hostmgr"
+)
+
+const (
+	baseConfig = `default_coreos_version: myversion
+network:
+  ip_range:
+    start: 1.1.1.1
+    end: 1.1.1.2
+templates_env:
+  mayu_https_endpoint: https://mayu
+`
+	configOK             = baseConfig + "  storage: mystorage"
+	configErr            = baseConfig
+	lastStageCloudconfig = `before_key: ok
+{{if eq .TemplatesEnv.storage "mystorage"}}storage: mystorage{{end}}
+after_key: ok
+`
+)
+
+var (
+	dir      string
+	fakeEtcd *httptest.Server
+	pxeCfg   PXEManagerConfiguration
+	req      *http.Request
+	w        *httptest.ResponseRecorder
+	cluster  *hostmgr.Cluster
+)
+
+func setUp(t *testing.T) {
+	var err error
+	dir, err = ioutil.TempDir("", "pxmgr_cloudconfig_")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err := ioutil.WriteFile(filepath.Join(dir, "config_ok.yaml"), []byte(configOK), 0644); err != nil {
+		t.Fatal(err)
+	}
+	if err := ioutil.WriteFile(filepath.Join(dir, "config_err.yaml"), []byte(configErr), 0644); err != nil {
+		t.Fatal(err)
+	}
+	if err := ioutil.WriteFile(filepath.Join(dir, "last_stage_cloudconfig.yaml"), []byte(lastStageCloudconfig), 0644); err != nil {
+		t.Fatal(err)
+	}
+
+	cluster, err = hostmgr.NewCluster(dir, true)
+	if err != nil {
+		t.Fatalf("creating cluster: %s", err)
+	}
+
+	fakeEtcd = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintln(w, "{}")
+	}))
+
+	pxeCfg = PXEManagerConfiguration{
+		UseInternalEtcdDiscovery: true,
+		NoTLS:                true,
+		APIPort:              4080,
+		PXEPort:              4081,
```

Comment:
In tests I would rather like to use random ports like the test server in golang generates. These tests can now not be executed in parallel and we should see a lot of "port already in use" errors.